mocks/tests/test_mocks.py

```diff
@@ -34,8 +34,120 @@ def pixelated_psf_data(huntsman_sbig_dark_imager):
     return psf_data
 
 
-def test_prepare_mocks():
-    mocks_config, data = mocks.prepare_mocks()
+def test_load_configuration(configuration):
+    assert configuration['data_path'] == 'sim_data/cl19.fits'
+    assert configuration['galaxy_distance'] == 10.
+    assert isinstance(configuration, dict)
+
+
+def test_parse_config():
+    config = mocks.parse_config()
+    assert isinstance(config, dict)
+    assert config['galaxy_coordinates'] == '14h40m56.435s -60d53m48.3s'
+    assert config['observation_time'] == '2018-04-12T08:00'
+    assert config['data_path'] == 'sim_data/cl19.fits'
+    assert config['imager_filter'] == 'g'
+    assert config['mass_to_light_ratio'] == {'g': 5, 'r': 5}
+    assert config['abs_mag_sun'] == {'g': 5.11, 'r': 4.68, 'i': 4.53}
+    assert config['galaxy_distance'].value == 10.
+    assert config['galaxy_distance'].unit == u.Mpc
+    assert config['target_galaxy_comoving_depth'].value == 1.
+    assert config['target_galaxy_comoving_depth'].unit == u.Mpc
+    assert config['viewing_axis'] == 'z'
+    assert config['sim_pc_pixel'].value == 170.
+    assert config['sim_pc_pixel'].unit == u.pc / u.pixel
+    assert config['particle_baryonic_mass_sim'] == 9.6031355
+    assert config['hubble_constant'].value == 70.5
+    assert config['hubble_constant'].unit == u.km / (u.Mpc * u.s)
+    assert config['Omega_0'] == 0.273
+    assert config['Omega_b'] == 0.04
+    assert config['Omega_Lambda'] == 0.727
+    assert config['T_CMB0'].value == 2.7
+    assert config['T_CMB0'].unit == u.K
+    assert config['Neff'] == 3.
+    assert config['m_nu'][2].value == 0.01
+    assert config['m_nu'].unit == u.eV
+
+
+def test_parse_config_default_cosmology():
+    config = mocks.parse_config('config_no_cosmology_example.yaml')
+    assert isinstance(config, dict)
+    assert config['hubble_constant'].value == 69.32
+    assert config['hubble_constant'].unit == u.km / (u.Mpc * u.s)
+    assert config['Omega_0'] == 0.2865
+    assert config['Omega_b'] == 0.04628
+    assert config['Omega_Lambda'] == 0.7135
+    assert config['T_CMB0'].value == 2.725
+    assert config['T_CMB0'].unit == u.K
+    assert config['Neff'] == 3.04
+    assert config['m_nu'].value.all() == 0
+    assert config['m_nu'].unit == u.eV
+
+
+def test_parse_config_none_value_cosmology_parameters():
```

Comment:
As notes in other comments, I don't think you need to allow for the possibility of keys without values in the config file. That's just a bad config file.